assets/chef-recipes/cookbooks/mariadb/recipes/install_community.rb

```diff
@@ -173,3 +173,10 @@
     command "echo '\n!includedir #{db_config_dir}' >> #{db_base_config}"
   end
 end
+
+directory db_config_dir do
```

Comment:
You must set permissions only to the configuration file that is being copied, not the whole directory! The permissions 644 is wrong for directories.